src/main/java/Measurement.java

```diff
@@ -1,19 +1,47 @@
 import org.json.JSONArray;
 import org.json.JSONObject;
 import java.util.ArrayList;
+import java.util.Date;
 import java.util.List;
-
+//this class will hold jobs
 public class Measurement{
-        static List<JSONObject> activeTask=new ArrayList<JSONObject>();
+   private static List<Job> activeJobs=new ArrayList<Job>();
 
-        public static boolean addMeasurement(JSONObject task){
-            if(task==null) return false;
-            //TODO need to add more error checking for valid task structure before addition
-            activeTask.add(task);
+   public static boolean addMeasurement(JSONObject jobRequest){
+            if(jobRequest==null) return false;
+            //TODO there is need for error checking for valid jobRequest structure before addition
+            JSONObject jobDesc = jobRequest.getJSONObject("job_description");
+            Job jobTobeScheduled=new Job(jobDesc);
+            activeJobs.add(jobTobeScheduled);
             return true;
         }
 
-        public static JSONArray getActiveTask(){
-            return new JSONArray(activeTask);
-        }
-}
+    public static JSONArray getActiveJobs() {
+            JSONArray sentJobs=new JSONArray();
+            Date currentTime=new Date();
+            for(Job job:activeJobs) {
+                if(job.canStart(currentTime)&&!job.nodesReached()){
+                    sentJobs.put(job.getMeasurementDesc());
+                }
+            }
+            return sentJobs;
+    }
+    //TODO reverse this to record successful as might remove it when some have failed
+    //and havent got the response yet
+    public static void recordSuccessfulJob(JSONObject jobDesc){
```

Comment:
Do you really need the JSONObject. Cant we just do with passing the job key?
Therefore we can identify jobs by their keys

Reply:
No we dont . I just needed to talk to you about it since i did not know how am going to receive the response from you

Reply:
@MelulekiDube pass a boolean and jobId so that i can mark a job as ```successful```